hooks unit test broken with /usr/bin/env from uutils coreutils (rust version)

On Ubuntu 26.04 gnu coreutils is mostly replaced by coreutils from the uutils project (a cross-platform reimplementation of the GNU coreutils in Rust).

A unit test symlinking `/usr/bin/env` to an arbitrary named executable behaves now different. Instead of just printing the environment, it tries to run the command from $0:
```
Running ganeti.hooks_unittest.py
......F........
======================================================================
FAIL: testEnv (__main__.TestHooksRunner.testEnv)
Test environment execution
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/gntbuild.45e2OvAV/test/py/legacy/ganeti.hooks_unittest.py", line 205, in testEnv
    self.assertEqual(self.hr.RunHooks(self.hpath, phase, env_snt),
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                         [(self._rname(fname), HKR_SUCCESS, env_exp)])
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Lists differ: [('fake-pre.d/success', 1, "coreutils: unknown program 'success'")] != [('fake-pre.d/success', 2, 'PHASE=pre')]

First differing element 0:
('fake-pre.d/success', 1, "coreutils: unknown program 'success'")
('fake-pre.d/success', 2, 'PHASE=pre')

- [('fake-pre.d/success', 1, "coreutils: unknown program 'success'")]
+ [('fake-pre.d/success', 2, 'PHASE=pre')]

```

--- a/test/py/legacy/ganeti.hooks_unittest.py
+++ b/test/py/legacy/ganeti.hooks_unittest.py
@@ -198,7 +198,10 @@ class TestHooksRunner(unittest.TestCase):
     for phase in (constants.HOOKS_PHASE_PRE, constants.HOOKS_PHASE_POST):
       fbase = "success"
       fname = "%s/%s" % (self.ph_dirs[phase], fbase)
-      os.symlink("/usr/bin/env", fname)
+      content = "#!/usr/bin/env sh\nenv --unset PWD\n"
+      with open(fname, "w", encoding="utf-8", newline="\n") as f:
+        f.write(content)
+      os.chmod(fname, 0o755)
       self.torm.append((fname, False))
       env_snt = {"PHASE": phase}
       env_exp = "PHASE=%s" % phase

